### ISSUE TITLE
How does resetOnError work?

### ISSUE THREAD

**Issue by Guito** [nuxt-community/auth-module#139]
The documentation says

resetOnError
Default: false
If enabled, user will be automatically logged out if any error happens. (For example when token expired)

Is this working? I have been checking the code and I don't see any code related to it

<div align="right"><sub><em>This question is available on <a href="https://nuxtjs.cmty.io">Nuxt.js</a> community (<a href="https://nuxtjs.cmty.io/nuxt-community/auth-module/issues/c104">#c104</a>)</em></sub></div>

**Comment by ishitatsuyuki** [comment]
@nitescent resetOnError is reimplemented. Please install directly from GitHub `dev` branch.

**Comment by nitescent** [comment]
@ishitatsuyuki It's not working on my side. Or it's unclear for me how to use the option. Would you have an example?

What I did is disable my plugin (see my previous post) and set auth {resetOnError: true} in nuxt.config.js then I login and make an other request wihch has a response from the server with status code 401 but nothing happens.

I used some log in @nuxtjs/auth/lib/core/auth.js to see what's happening. 'log1' and 'log2' are printed but 'log3' is not printing when server returns a 401.

```js
async init () {
    console.debug('log1');
    // Reset on error
    if (this.options.resetOnError) {
      console.debug('log2');
      this.onError((...args) => {
        console.debug('log3');
        if (typeof (this.options.resetOnError) !== 'function' || this.options.resetOnError(...args)) {
          this.reset()
        }
      })
    }
```

**Comment by b4dnewz** [comment]
Why no updates on this issue?

nuxt ^2.4.5
nuxtjs/auth ^4.5.3

Still not working.

**Comment by saschametz** [comment]
I'm have the same issue. `resetOnError` does not redirect to login if the token is expired.

**Comment by jlowgren** [comment]
I'd also love for this to be sorted out.

```
nuxt@2.8.1
@nuxtjs/auth@4.7.0
```

**Comment by andreynazarov3** [comment]
hello, i see that `resetOnError` works, but, it work only after 2 page refreshes. After 1st auth error it shows 401 error page and sets unlogged cookies, so after 2nd refresh page loades without problems...

**Comment by jarkt** [comment]
I have the same behaviour like @andreynazarov3 
Yes, the option leads to a different result and the token is removed after 2 page reloads... but that's not the expected behaviour.
Currently I don't know how to implement the requirement without workaround.

The issue is labeled as question, but it looks like a bug.

**Comment by andreynazarov3** [comment]
> 
> 
> I have the same behaviour like @andreynazarov3
> Yes, the option leads to a different result and the token is removed after 2 page reloads... but that's not the expected behaviour.
> Currently I don't know how to implement the requirement without workaround.
> 
> The issue is labeled as question, but it looks like a bug.

seems this property works only if redirect on logout is set:
```
 redirect: {
      // redirect here after logout
      logout: 'index'
}
```

**Comment by jarkt** [comment]
Thx, but that seems not to be enough. This is my config:

```
auth: {
    strategies: {
      local: {
        endpoints: {
          login: { url: '/api/login_check', method: 'post', propertyName: 'token' },
          // logout: { url: '/api/auth/logout', method: 'post' },
          user: { url: '/api/user', method: 'get', propertyName: '' }
        }
      }
    },
    redirect: {
      login: '/login',
      home: '/',
      logout: '/'
    },
    resetOnError: true
  }
```

BTW: "index" would not be work

**Comment by andreynazarov3** [comment]
yeah, i use 'index' because i have localize routes before redirect.
i notice one difference with my config that i have logout endpoint to false:
```
strategies: {
      local: {
        endpoints: {
          login: {
            url: '/shop-api/login_check',
            method: 'post',
            propertyName: 'token'
          },
          logout: false,
          user: {
            url: '/shop-api/me',
            method: 'get',
            propertyName: false
          }
        }
```

**Comment by jarkt** [comment]
Unfortunately that makes also no difference.

**Comment by artmadeit** [comment]
Currently not working :(

**Comment by enriquesoto** [comment]
not working here :cry:

**Comment by andreynazarov3** [comment]
guys, i found what workaround i used for it
in plugins/axios.js i set special case for 401 error like
```
$axios.interceptors.response.use(
    function(response) {
      return response;
    },
    function(error) {
      const code = parseInt(error.response && error.response.status);

      if ([401, 403].includes(code)) {
        app.$auth.logout();
      }

      return Promise.reject(error);
    }
  );
```

**Comment by Bergrebell** [comment]
still having this error ... thanks for the workaround @andreynazarov3

**Comment by louisluu** [comment]
Tried setting `resetOnError: true` and nothing happened.

According to the [document](https://auth.nuxtjs.org/api/options.html#resetonerror), it accepts **function** as well, so tried:

```js
// nuxt.config.js
{
  auth: {
   // other configs...
   resetOnError() {
      console.log('here 123')
      return true
    },
  },
}
```

When 401 happens, nothing is printed in the console. 

Searching the source code and found its used in https://github.com/nuxt-community/auth-module/blob/dev/src/core/auth.ts#L35 but I tried setting up auth plugin to test the `onError` like this:

```js
// plugins/auth.js
export default ({ $auth }) => {
  console.log('auth plugin')
  $auth.onError((error, name, endpoint) => {
    console.log('')
    console.log('Auth error')
    console.log('error', error)
    console.log('name', name)
    console.log('endpoint', endpoint)
  })
}
```

Can see **auth plugin** printed in console but when 401 happens, nothing is printed.

---

Seems like the problem is from the `onError` ?!?

**Comment by bmulholland** [comment]
Closing as stale; please open a new issue if this is still happening.
